Compilation errors with Slurm 20.11

Hi!

Just tried the last versions of spart, and noticed the following compilation errors with CentOS/Red Hat 7.x default gcc version (4.8.5):

```
# rpmbuild -ta spart-1.4.3.tar.gz 
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.zku8Ug
+ umask 022
+ cd /root/rpmbuild/BUILD
+ cd /root/rpmbuild/BUILD
+ rm -rf spart-1.4.3
+ /usr/bin/gzip -dc /root/rpmbuild/SOURCES/spart-1.4.3.tar.gz
+ /usr/bin/tar -xf -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd spart-1.4.3
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.hTfs7J
+ umask 022
+ cd /root/rpmbuild/BUILD
+ cd spart-1.4.3
+ gcc -lslurm spart.c -o spart
spart.c: In function 'main':
spart.c:29:3: error: unknown type name 'slurm_ctl_conf_t'
   slurm_ctl_conf_t *conf_info_msg_ptr = NULL;
   ^
spart.c:290:3: warning: passing argument 2 of 'slurm_load_ctl_conf' from incompatible pointer type [enabled by default]
   if (slurm_load_ctl_conf((time_t)NULL, &conf_info_msg_ptr)) {
   ^
In file included from spart.c:11:0:
/usr/include/slurm/slurm.h:3714:12: note: expected 'struct slurm_conf_t **' but argument is of type 'int **'
 extern int slurm_load_ctl_conf(time_t update_time,
            ^
spart.c:312:32: error: request for member 'cluster_name' in something not a structure or union
               conf_info_msg_ptr->cluster_name, SPART_MAX_COLUMN_SIZE);
                                ^
spart.c:316:24: error: request for member 'private_data' in something not a structure or union
   if (conf_info_msg_ptr->private_data != 0) {
                        ^
spart.c:320:26: error: request for member 'private_data' in something not a structure or union
     if (conf_info_msg_ptr->private_data & PRIVATE_DATA_JOBS) {
                          ^
spart.c:328:26: error: request for member 'private_data' in something not a structure or union
     if (conf_info_msg_ptr->private_data & PRIVATE_DATA_NODES) {
                          ^
spart.c:332:26: error: request for member 'private_data' in something not a structure or union
     if (conf_info_msg_ptr->private_data & PRIVATE_DATA_PARTITIONS) {
                          ^
spart.c:342:3: error: too few arguments to function 'slurmdb_connection_get'
   db_conn = slurmdb_connection_get();
   ^
In file included from spart.c:13:0:
/usr/include/slurm/slurmdb.h:1555:14: note: declared here
 extern void *slurmdb_connection_get(uint16_t *persist_conn_flags);
              ^
spart.c:386:14: warning: assignment makes pointer from integer without a cast [enabled by default]
         qos2 = slurmdb_qos_str(qosn_list, atoi(qos));
              ^
spart.c:1372:3: warning: passing argument 1 of 'slurm_free_ctl_conf' from incompatible pointer type [enabled by default]
   slurm_free_ctl_conf(conf_info_msg_ptr);
   ^
In file included from spart.c:11:0:
/usr/include/slurm/slurm.h:3722:13: note: expected 'struct slurm_conf_t *' but argument is of type 'int *'
 extern void slurm_free_ctl_conf(slurm_conf_t *slurm_ctl_conf_ptr);
             ^
error: Bad exit status from /var/tmp/rpm-tmp.hTfs7J (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.hTfs7J (%build)
```

--- a/spart.c
+++ b/spart.c
@@ -26,7 +26,11 @@ int main(int argc, char *argv[]) {
   int total_width = 0;
 
   /* Slurm defined types */
+#if SLURM_VERSION_NUMBER >= SLURM_VERSION_NUM(20, 11, 0)
+  slurm_conf_t *conf_info_msg_ptr = NULL;
+#else
   slurm_ctl_conf_t *conf_info_msg_ptr = NULL;
+#endif
   partition_info_msg_t *part_buffer_ptr = NULL;
   partition_info_t *part_ptr = NULL;
   node_info_msg_t *node_buffer_ptr = NULL;
@@ -339,7 +343,11 @@ int main(int argc, char *argv[]) {
     SLURM_VERSION_NUMBER != SLURM_VERSION_NUM(20, 2, 0) && \
     SLURM_VERSION_NUMBER != SLURM_VERSION_NUM(20, 2, 1)
   /* Getting user account info */
+#if SLURM_VERSION_NUMBER >= SLURM_VERSION_NUM(20, 11, 0)
+  db_conn = slurmdb_connection_get(NULL);
+#else
   db_conn = slurmdb_connection_get();
+#endif
   if (errno != SLURM_SUCCESS) {
     slurm_perror("Can not connect to the slurm database");
     exit(1);

